File: Semester 4/OOP/Praktikum/Praktikum 2/Complex.cpp
#include "Complex.h"

using namespace std;

Complex::Complex(){
	real=0;
	imaginer=0;
}

Complex::Complex(int re, int im){
	real = re;
	imaginer = im;
}

Complex::Complex(const Complex &comp){
	this->real = comp.real;
	this->imaginer = comp.imaginer;
}

Complex::~Complex(){
	
}

int Complex::getReal(){
	return real;
}
	
int Complex::getImaginer(){
	return imaginer;
}

void Complex::setReal(int re){
	real = re;
}

void Complex::setImaginer(int im){
	imaginer = im;
}

Complex Complex::operator +(Complex const &comp){
	Complex t(real+comp.real, imaginer+comp.imaginer);
	return t;
}

Complex Complex::operator -(Complex const &comp){
	Complex t(real-comp.real, imaginer-comp.imaginer);
	return t;
}

Complex Complex::operator *(Complex const &comp){
	Complex t(real*comp.real - imaginer*comp.imaginer, imaginer*comp.real + real*comp.imaginer);
	return t;
}

File: Semester 4/OOP/Praktikum/Praktikum 9/Complex.java
class Complex{
	private int real;
	private int imaginer;
	
	public Complex(){
		real = imaginer = 0;
	}
	public Complex(int a, int b){
		real = a;
		imaginer = b;
	}
	public int getReal(){
		return real;
	}
	public int getImaginer(){
		return imaginer;
	}
	public void setReal(int a){
		real = a;
	}
	public void setImaginer(int b){
		imaginer = b;
	}
	
	// Mengembalikan objek Complex dengan nilai ini ditambah b
	public Complex plus(Complex b){
		Complex ans = new Complex(real + b.real, imaginer+b.imaginer);
		return ans;
	}

	// Mengembalikan objek Complex dengan nilai ini dikurangi b
	public Complex minus(Complex b){
		Complex ans = new Complex(real - b.real, imaginer - b.imaginer);
		return ans;
	}

	// Mengembalikan objek Complex dengan nilai ini dikali b
	public Complex multiply(Complex b){
		Complex ans = new Complex(real*b.real - imaginer*b.imaginer, imaginer*b.real + real*b.imaginer);
		return ans;
	}
}
